### DISCUSSION TITLE
Using Runout Sensor Port to Power LED

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1683] [General Discussion]
Hi

I'm trying to add LEDs to illuminate the nozzle and I have these bright 5V USB LED (0.2W) chips and I'm thinking to use the runout sensor port on the 4.2.2 board as a power source. Some sites mention that the port supplies 5V and others 3.3V but anyway I have tested the LEDs with 3.3V and they are still very bright.

I was wondering if anyone has tried using this port for similar applications as I couldn't find info anywhere online. The LEDs draw 0.2W only so is this ok ?

Thanks

![voltaat-0-2-w-22-lumen-mini-usb-led-module-15154481332326_1024x1024](https://user-images.githubusercontent.com/65966061/145724823-d6e18cde-b7db-471d-aabe-913f6fd19e27.jpg)

![voltaat-0-2-w-22-lumen-mini-usb-led-module-28338312216678_480x480](https://user-images.githubusercontent.com/65966061/145724920-9cd84640-0cc5-48e0-9c69-6e5c29786c1f.jpg)

![Layout 1](https://user-images.githubusercontent.com/65966061/145724834-81d4e2ac-5655-4790-8aa6-cc7b5b6be0d6.jpeg)

**Comment by Cal-LF** [comment]
Not sure what the current rating is for the filament port but 200ma required by the LED may be too much and blow the main fuse F1 or second fuse F2. Just an FYI for you.

**Reply by BlueWings172** [reply]
I decided to take the risk and try it and while there was no magic smoke, the LEDs barely emitted any light as you mentioned.

Is there any other solution to get 5V or 3V directly from the board to power the LEDs without buck converters ?

**Comment by Cal-LF** [comment]
Looking at the schematic, I would say there is no 3v or 5v source to provide enough current and allow the main board to function reliably.

**Comment by BlueWings172** [comment]
Does the 4.2.2 board USB provide power 5V power ?

If yes, is there anyway I can solder a wire to the board pin giving the USB it's 5V while keep using the USB port for data to communicate with Octoprint ?

**Comment by Cal-LF** [comment]
Yes and the 3.3v is derived from the 5v. Look at the 24v input and follow to the U1 processor which seems to output the 5v. I recommend to not use the main board as a power source for anything except 24v terminals. The USB port uses a protection diode and current may be limited.
[Creality.4.2.2.-.Schematic.22-9-21.pdf](https://github.com/Jyers/Marlin/files/7713928/Creality.4.2.2.-.Schematic.22-9-21.pdf)

I use and external 4 port USB power outlet for my lighting. 3 clamp type reading lamps. I then have a remote switched 120 v control box for the USB outlet

![Image01](https://user-images.githubusercontent.com/76499230/146061238-7174dc04-ba47-4a1b-9f5d-95e3e9fd1187.jpg)
![Image02](https://user-images.githubusercontent.com/76499230/146061243-f9420afc-da8a-4bf5-9cce-71e5e6a490a7.jpg)
.

**Comment by BlueWings172** [comment]
the pins going to the USB are way too thin to be soldered so that's a dead-end.

However, I think I found the perfect solution. Looking more closely at the board photos and the schematics you shared, I found the ICSP port that can provide 3.3V with enough current for the LEDs. Not only that, but there are pin headers already there and I don't need even to solder anything to the board.

I have used Dupont connectors to connect VCC and ground to two wires and soldered the other end to the USB LED chip and the light was really bright.  

Below are some photos and I will be designing or remixing a fan shroud to include a place for the chip so I don't have to glue it and have close the height of the nozzle.

I don't know the long term impact of this solution on the board and I haven't tested yet if the lights will be as bright when the printer is printing.

Will test and report.

![ICSP Port](https://user-images.githubusercontent.com/65966061/146204126-9016318c-ff75-4f19-bd14-b0371ff9aa77.jpg)


![Pins 1](https://user-images.githubusercontent.com/65966061/146204252-7d86831b-c9f7-4d1d-8e3b-f3bf714432d0.jpg)


![Pins 2](https://user-images.githubusercontent.com/65966061/146204194-ed333c1e-d9ca-4b0b-b00d-4661853480b7.jpg)


![Pins 3](https://user-images.githubusercontent.com/65966061/146204297-a85f54eb-6d64-4ab0-8681-44285408013f.jpg)

![20211215_171526](https://user-images.githubusercontent.com/65966061/146204338-e144795f-5f66-4354-a925-8aa5c1efc03b.jpg)

![20211215_140958](https://user-images.githubusercontent.com/65966061/146204430-864f324d-40dc-4514-b7c3-76b2063daf3f.jpg)


![20211215_141147](https://user-images.githubusercontent.com/65966061/146204529-d44236bf-81f9-44c5-8f78-ecd037ef1668.jpg)


![20211215_141214](https://user-images.githubusercontent.com/65966061/146204654-505f00c5-cf50-4e0d-89c5-9e38688fca9b.jpg)


![20211215_141223](https://user-images.githubusercontent.com/65966061/146204680-c5254023-84c9-40e4-aae6-d6b9b91461ac.jpg)

**Comment by Cal-LF** [comment]
@BlueWings172 Looks like a successful upgrade. You have skill in electronics and mechanical.

**Reply by BlueWings172** [reply]
Thanks man. I'm actually new to all this. Before COVID, I never thought of buying a 3D printer or using a multimeter.

Hopefully this helps someone.

**Comment by Cal-LF** [comment]
@BlueWings172 I see the led lamp is 5vdc @ 40ma, not 200ma as I stated above. That amount of current should not cause an issue.
